src/components/search/search-result.js

```diff
@@ -0,0 +1,219 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { injectIntl } from 'react-intl'
+import { withRouter } from 'react-router'
+import queryString from 'query-string'
+import deepEqual from 'deep-equal'
+import $ from 'jquery'
+import 'rc-slider/assets/index.css'
+
+import schemaMessages from '../../schemaMessages/index'
+import { showAlert } from 'actions/Alert'
+import ListingsGrid from 'components/listings-grid'
+import SearchBar from 'components/search/searchbar'
+import { generalSearch } from 'actions/Search'
+import origin from '../../services/origin'
+import FilterGroup from 'components/search/filter-group'
+
+class SearchResult extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      filterSchema: undefined,
+      listingSchema: undefined,
+      listingType: undefined,
+      listingIds: [],
+      searchError: undefined,
+      filters: {}
+    }
+
+    // set default prop values for search_query and listing_type
+    const getParams = queryString.parse(this.props.location.search)
+    this.props.generalSearch(getParams.search_query || '', getParams.listing_type || 'all', false, false)
+  }
+
+  shouldFetchListingSchema() {
+    return this.props.listingType !== 'all'
+  }
+
+  componentDidMount() {
+    /* this force update is required after component initializes. In cases where user returns to the 
+     * search-result page. Then no props change from previous to current and for that reason search and
+     * schema loading do not get triggered.
+     */
+    this.handleComponentUpdate(undefined)
+
+    // Keep dropdown opened when user clicks on any element in the dropdownw
+    $(document).on('click', '#search-filters-bar .dropdown-menu', e => {
```

Comment:
Can this be accomplished using [a dropdown event listener](https://getbootstrap.com/docs/4.0/components/dropdowns/#events)? Maybe not. How about an `onClick` handler on the relevant React components? I would like to get away from jQuery at some point, but it doesn't have to start here.

Reply:
Afaik returning false from `hide.bs.dropdown ` event should stop the dropdown from closing. But you would still need to use jQuery to attach to that event. I am not happy with this solution also, but couldn't find a better one.